pkg/trace/api/api.go

```diff
@@ -546,6 +538,10 @@ func (r *HTTPReceiver) handleTraces(v Version, w http.ResponseWriter, req *http.
 		log.Errorf("Cannot decode %s traces payload: %v", v, err)
 		return
 	}
+	defer func(err error) {
```

Comment:
I think you need to place the func below line 523, otherwise the return on line 539 could cause the metric not to be reported, specifically in an interesting case when an error occurs